### PR TITLE
Proxy & log speech-to-speech sessions on OpenAI Realtime API.

### PR DESCRIPTION
**Pull request by kevin1**

### CODE REVIEW

File: apis/cloudflare/src/realtime-logger.ts [hunk 1]
```diff
@@ -0,0 +1,772 @@
...
+// - Where the OpenAI implementation differs from their type spec.
+// - Replace the fields we don't use with `z.any()` for more permissive parsing.
+const baseMessageSchema = z.object({
+  event_id: z.string(),
+});
+
+const audioFormatTypeSchema = z.enum(["pcm16", "g711_ulaw", "g711_alaw"]);
+type AudioFormatType = z.infer<typeof audioFormatTypeSchema>;
+
+const turnDetectionServerVadTypeSchema = z.object({
```
Comment (ankrgyl) [inline apis/cloudflare/src/realtime-logger.ts:19]: small nit: split the schema definitions into a separate file

File: apis/cloudflare/src/realtime-logger.ts [hunk 2]
```diff
@@ -0,0 +1,772 @@
...
+        algorithm: "a",
+        sample_rate: 8000,
+      };
+    default:
+      const x: never = audioFormat;
+      throw new Error(`Unknown audio format ${JSON.stringify(x)}`);
+  }
+}
+
+/**
```
Comment (ankrgyl) [inline apis/cloudflare/src/realtime-logger.ts:464]: i wonder if we shuold move this into the SDK (perhaps in a separate change). So you can use it outside of the proxy too 🤔

**Review by ankrgyl — APPROVED** [review]

File: apis/cloudflare/src/realtime-logger.ts (hunk 1 above)
Comment (kevin1) [reply apis/cloudflare/src/realtime-logger.ts:19]: Done

File: apis/cloudflare/src/realtime-logger.ts (hunk 2 above)
Comment (kevin1) [reply apis/cloudflare/src/realtime-logger.ts:464]: Will do in a separate change. Probably also needs some API cleanup before we can commit to maintaining it as part of the SDK.

**Pull request merged**